tests/flow/test_concurrent_query.py

```diff
@@ -277,3 +277,54 @@ def test_09_concurrent_multiple_readers_after_big_write(self):
                 self.env.assertEquals(0, result)
             else:
                 self.env.assertEquals(1000, result["result_set"][0][0])
+
+    def test_10_write_starvation(self):
+        # make sure write query do not starve
+        # when issuing a large number of read queries
+        # alongside a single write query
+        # we dont want the write query to have to wait for
+        # too long, consider the following sequence:
+        # R, W, R, R, R, R, R, R, R...
+        # if write is starved our write query might have to wait
+        # for all queued read queries to complete while holding
+        # Redis global lock, this will hurt performance
+        #
+        # this test issues a similar sequence of queries and
+        # validates that the write query wasn't delayed too much
+
+        self.graph = Graph(GRAPH_ID, self.conn)
+        Rq = "UNWIND range(0, 10000) AS x WITH x WHERE x = 9999 RETURN 'R', timestamp()"
+        Wq = "UNWIND range(0, 1000) AS x WITH x WHERE x = 27 CREATE ({v:1}) RETURN 'W', timestamp()"
+
+        # create a long sequence of read queries
+        queries = [Rq] * CLIENT_COUNT * 10
+        nulls = [None] * CLIENT_COUNT * 10
+
+        # inject a single write query close to the begining on the sequence
+        queries[CLIENT_COUNT] = Wq
+
+        # invoke queries
+        # execute queries in parallel
+        pool = Pool(nodes=CLIENT_COUNT)
+        results = pool.map(thread_run_query, queries, nulls)
+
+        # search for 'W' position
+        # collect each query timestamp
+        write_ts = None
```

Comment:
because you already know the index for the write query maybe we can only count the number of queries with less than timestamp?
I think this is a better check